cachito/workers/pkg_managers/pip.py

```diff
@@ -1921,3 +1926,132 @@ def get_index_url(nexus_pypi_hosted_repo_url, username, password):
         )
 
     return index_url
+
+
+def _download_from_requirement_files(request_id, files):
+    """
+    Download dependencies listed in the requirement files.
+
+    :param int request_id: ID of the request these dependencies are being downloaded for
+    :param list files: list of str, each representing the absolute path of a pip requirement file
+    :return: Info about downloaded packages; see download_dependencies return docs for further
+        reference
+    :rtype: list[dict]
+    """
+    requirements = []
+    for req_file in files:
+        requirements.extend(download_dependencies(request_id, PipRequirementsFile(req_file)))
+    return requirements
+
+
+def _default_requirement_file_list(path, devel=False):
+    """
+    Get the paths for the default pip requirement files, if they are present.
+
+    :param Path path: the full path to the application source code
+    :param bool devel: whether to return the build requirement files
+    :return: list of str representing the absolute paths to the Python requirement files
+    :rtype: list[str]
+    """
+    filename = DEFAULT_BUILD_REQUIREMENTS_FILE if devel else DEFAULT_REQUIREMENTS_FILE
+    req = path / filename
+    return [str(req)] if req.is_file() else []
+
+
+def _push_downloaded_requirement(requirement, pip_repo_name, raw_repo_name):
+    """
+    Upload a dependency to the proper Request temporary Nexus repository.
+
+    :param dict requirement: Single entry with the info about downloaded packages retrieved from the
+        list returned by the download_dependencies function
+    :param str pip_repo_name: Name of the Nexus Python hosted repository to push the requirement to
+        (if pypi)
+    :param str raw_repo_name: Name of the Nexus raw hosted repository to push the requirement to
+        (if not pypi)
+    :return: dict with the cachito Dependency representation
+    :rtype: dict
+    """
+    if requirement["kind"] == "pypi":
+        upload_pypi_package(pip_repo_name, requirement["path"])
+        dep = {"name": requirement["package"], "version": requirement["version"], "type": "pip"}
+    else:
+        dest_dir, filename = requirement["raw_component_name"].rsplit("/", 1)
+        upload_raw_package(raw_repo_name, requirement["path"], dest_dir, filename, True)
+        version = (
+            f"vcs:{requirement['host']}/{requirement['namespace']}/{requirement['repo']}@"
+            f"{requirement['ref']}"
+        )
```

Comment:
Now that #255 is merged, do you want to add "url" handling here? Or in another PR? Currently, url dependencies woud fail here with a `KeyError: 'host'`